File: Backend/src/main/java/com/example/CheckInApp/dto/response/EventResponse.java
package com.example.CheckInApp.dto.response;

import com.example.CheckInApp.model.EventLocation;
import com.example.CheckInApp.model.EventStatus;
import com.example.CheckInApp.model.EventType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class EventResponse {
    private Long id;
    private String name;
    private EventLocation location;
    private Instant startDateTime;
    private Instant endDateTime;
    private EventType type;
    private EventStatus status;
    private String poster;
    private Instant registrationStartDate;
    private Instant registrationEndDate;
    private String address;
    private String description;
    private Long createdById;
    private Boolean foodProvided;
    private Instant createdAt;
    private boolean codesGenerated;
    private boolean userRegistered;
    private boolean userCheckedIn;
}

File: Frontend/src/app/core/events/dto/event.response.ts
import { EventLocation } from '@core/events/model/event-location';
import { EventStatus } from '@core/events/model/event-status';
import { EventType } from '@core/events/model/event-type';

export type EventResponse = {
  id: number;
  name: string;
  location: EventLocation;
  startDateTime: string;
  endDateTime: string;
  type: EventType;
  status: EventStatus;
  poster: string | null;
  registrationStartDate: string;
  registrationEndDate: string;
  address: string;
  description: string;
  createdById: number | null;
  foodProvided: boolean | null;
  createdAt: string;
};
